pydoctor/sphinx_ext/help_output.py

```diff
@@ -0,0 +1,53 @@
+"""
+Private extension that produces the pydoctor help output to be included in the documentation.
+"""
+from docutils import nodes
+from docutils.parsers.rst import Directive
+
+from contextlib import redirect_stdout
+from io import StringIO
+from typing import Any, Dict, List, TYPE_CHECKING
+
+from pydoctor.driver import parse_args
+
+
+if TYPE_CHECKING:
+    from sphinx.application import Sphinx
+
+
+class HelpOutputDirective(Directive):
+    """
+    Directive that will generate the pydoctor help as block literal.
+
+    It takes no options or input value.
+    """
+    has_content = True
+
+    def run(self) -> List[nodes.Node]:
+        """
+        Called by docutils each time the directive is found.
+        """
+
+        stream = StringIO()
+        try:
+            with redirect_stdout(stream):
+                parse_args(['--help'])
+        except SystemExit:
+            # The stdlib --help handling triggers system exit.
+            pass
+
+        text = ['pydoctor --help'] + stream.getvalue().splitlines()[1:]
```

Comment:
Let me know if you want unit tests for this.

Reply:
Maybe a system test is more appropriate: run the extension under Sphinx, check the exit code and check the presence of a few options that are unlikely to ever disappear in the output.

In any case, I don't think it needs detailed functional tests, but it would be good to execute it at least once as part of a CI cycle.

Reply:
OK. I am building the whole sphinx docs and then run a grep on the help.html page.